<?xml version="1.0"?>
<launch>
  <!-- Command-line arguments -->
  <arg name="ns" default="ifm3d"/>
  <arg name="nn" default="camera"/>
  <arg name="ip" default="192.168.0.69"/>
  <arg name="xmlrpc_port" default="80"/>
  <arg name="password" default=""/>
  <arg name="schema_mask" default="15"/>
  <arg name="timeout_millis" default="500"/>
  <arg name="timeout_tolerance_secs" default="5.0"/>
  <arg name="frame_id_base" default="$(arg ns)/$(arg nn)" />
  <arg name="respawn" default="false"/>
  <arg name="assume_sw_triggered" default="false"/>
  <arg name="sync_clocks" default="false"/>

  <include ns="$(arg ns)" file="$(find ifm3d)/launch/nodelet.launch">
    <arg name="camera" value="$(arg nn)"/>
    <arg name="ip" value="$(arg ip)"/>
    <arg name="xmlrpc_port" value="$(arg xmlrpc_port)"/>
    <arg name="password" value="$(arg password)"/>
    <arg name="schema_mask" value="$(arg schema_mask)"/>
    <arg name="timeout_millis" value="$(arg timeout_millis)"/>
    <arg name="timeout_tolerance_secs" value="$(arg timeout_tolerance_secs)"/>
    <arg name="frame_id_base" value="$(arg ns)/$(arg nn)" />
    <arg name="respawn" value="$(arg respawn)"/>
    <arg name="assume_sw_triggered" value="$(arg assume_sw_triggered)"/>
    <arg name="sync_clocks" value="$(arg sync_clocks)"/>
  </include>

</launch>
